registry/txt.go

```diff
@@ -138,11 +135,8 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 		if err != nil {
 			return nil, err
 		}
-		endpointName, isAAAA := im.mapper.toEndpointName(record.DNSName)
-		key := fmt.Sprintf("%s::%s", endpointName, record.SetIdentifier)
-		if isAAAA {
-			key += keySuffixAAAA
-		}
+		endpointName, recordType := im.mapper.toEndpointName(record.DNSName)
+		key := createKey(endpointName, record.SetIdentifier, recordType)
```

Comment:
If you want to simplify that, make the hash key a `struct`, possibly an `EndpointKey`. Don't serialize it to a `string`.

Reply:
It was a convoluted string join before, I made it slightly nicer. Bear in mind that this needs to work for records with both the old and new type txt records. String is the obvious choice without having to implement any additional logic, as you can check both example.com::A and example.com very easily https://github.com/kubernetes-sigs/external-dns/blob/74be95aca45d2c195b15a7e85bf62f269224fb78/registry/txt.go#L159. I don't really want to do anything, if you have a better solution then go for it.

Reply:
`EndpointKey` would also work for both old and new. For old, one would leave the `RecordType` field empty.

Reply:
Implemented